hub/values.yaml

```diff
@@ -180,6 +247,59 @@ jupyterhub:
             elif is_instructor:
               self.user.admin = True
 
+            ## custom.canvas_courses
+            # retrieve the user's enrollments from bcourses
+            canvas_courses = []
+            auth_state = yield self.user.get_auth_state()
+            if auth_state and 'access_token' in auth_state:
+              token = auth_state['access_token']
+              # FIXME: check token expiration rather than using try/except
+              try:
+                canvas_courses = yield self.get_canvas_courses(token)
+              except InvalidAccessToken as e:
+                self.log.warning(f"got invalid access token")
+                new_token = yield self.refresh_access_token(auth_state)
+                canvas_courses = yield self.get_canvas_courses(new_token)
+              except Exception as e:
+                self.log.info("error getting canvas courses")
+                self.log.info(str(e))
+            student_courses = course_filter(canvas_courses, 'student') + \
+                              course_filter(canvas_courses, 'observer')
+            instructor_courses = course_filter(canvas_courses, 'ta') + \
+                                 course_filter(canvas_courses, 'teacher')
+            self.log.info(f"student_courses: {student_courses}")
+            self.log.info(f"instructor_courses: {instructor_courses}")
+
+            # set resource limits, add volumes, set/revoke admin based on
+            # courses
+            custom_canvas_courses = z2jh.get_config('custom.canvas_courses', {})
+            is_canvas_student = False ; is_canvas_instructor = False
+            for canvas_course_id, profile_data in custom_canvas_courses.items():
+              self.log.warning(f'evaluating course id {canvas_course_id}')
+              is_canvas_student    |= (int(canvas_course_id) in student_courses)
```

Comment:
In general I try to avoid bitwise operations wherever possible in Python except for actual bitwise manipulation. Can this just be:
```
is_canvas_student = int(canvas_course_id) in student_courses
```

since we only care about setting it to true at least once.

Reply:
But that would turn it off if the user wasn't a student in the next course. I can replace the bitwise operator with something more verbose though.